### PR TITLE
Improved input overflow efficiency

### PR DESCRIPTION
**Pull request by wistpotion**
Rewrote an inefficient piece of code to be a little faster. The loops usually runs once each compared to the old code where one letters was removed at a time until the text fit. The new code is a little more complex, but it should save a little in performance. 
This performance boost doesn't really matter, but I didn't have anything better to do.

### CODE REVIEW

File: gooey/internal/input.lua [hunk 1]
```diff
@@ -89,13 +89,24 @@ function INPUT.set_text(input, text)
...
 
 		-- prevent text from overflowing the input field
-		-- this will get increasingly expensive the longer the text
-		-- future improvement is to make a best guess and then either add or
-		-- remove characters depending on if the text is too short or long
 		local field_width = gui.get_size(input.node).x * gui.get_scale(input.node).x
-		while (text_width + marked_text_width) > field_width do
-			text = text:sub(2)
-			text_width = get_text_width(input.node, text)
+		if (text_width + marked_text_width) > field_width then
```
Comment (britzl) [inline gooey/internal/input.lua:93]: Please add a comment describing the new approach!

**Review by britzl — CHANGES_REQUESTED** [review]
Looks good, but I think it could use a description of how the new text is calculated

File: gooey/internal/input.lua (hunk 1 above)
Comment (wistpotion) [reply gooey/internal/input.lua:93]: New commit should be better commented :D

**Review by britzl — APPROVED** [review]
Looks good!

**Pull request merged**